packages/sdk/tests/e2e/sharing/setProtectedDataAsRentable.test.ts

```diff
@@ -0,0 +1,120 @@
+  });
+
+  describe('When calling setProtectedDataToRenting()', () => {
+    it(
+      'should answer with success true',
+      async () => {
+        //Create a Protected data
+        const result = await dataProtector.protectData({
+          name: 'test',
+          data: { doNotUse: 'test' },
+        });
+        //create collection
+        const { collectionTokenId } = await dataProtector.createCollection();
+
+        const onStatusUpdateMock = jest.fn();
+        //add Protected Data To Collection
+        await dataProtector.addToCollection({
+          protectedDataAddress: result.address,
+          collectionTokenId,
+          onStatusUpdate: onStatusUpdateMock,
+        });
+        //Test price and duration values
+        const price = BigInt('100');
+        const duration = 2000;
+
+        const { success } = await dataProtector.setProtectedDataToRenting({
+          protectedDataAddress: result.address,
+          collectionTokenId: collectionTokenId,
+          durationInSeconds: duration,
+          priceInNRLC: price,
+        });
+        expect(success).toBe(true);
+      },
+      8 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
+    );
+    it(
+      'should fail with not collection owner error',
+      async () => {
+        //Create a Protected data
+        const result = await dataProtector.protectData({
+          name: 'test',
+          data: { doNotUse: 'test' },
+        });
+        //create collection
+        const { collectionTokenId } = await dataProtector.createCollection();
+
+        const onStatusUpdateMock = jest.fn();
+        //add Protected Data To Collection
+        await dataProtector.addToCollection({
+          protectedDataAddress: result.address,
+          collectionTokenId,
+          onStatusUpdate: onStatusUpdateMock,
+        });
+        const wallet1 = Wallet.createRandom();
+        const dataProtector1 = new IExecDataProtector(
+          getWeb3Provider(wallet1.privateKey)
+        );
+        //Test price and duration values
+        const price = BigInt('100');
+        const duration = 2000;
+
+        await expect(() =>
+          dataProtector1.setProtectedDataToRenting({
+            protectedDataAddress: result.address,
+            collectionTokenId: collectionTokenId,
+            durationInSeconds: duration,
+            priceInNRLC: price,
+          })
+        ).rejects.toThrow(
+          new WorkflowError(
+            'Failed to Set Protected Data To Renting: user is not collection owner.'
+          )
+        );
+      },
+      8 * MAX_EXPECTED_BLOCKTIME + MAX_EXPECTED_WEB2_SERVICES_TIME
+    );
+    it(
+      'should fail with protected data is not in collection error',
+      async () => {
+        const protectedDataAddressMock = Wallet.createRandom().address;
+        //create collection
+        const { collectionTokenId } = await dataProtector.createCollection();
```

Comment:
After this call we need to wait a few seconds for the subgraph to have time to index the event.
```
await sleep(2_000);
```

? 🤷‍♂️

Or setting this value as a constant as we'll reuse it in some other tests.
```
export const SUBGRAPH_EVENT_INDEXING_WAIT_TIME = 80_000;
```